simulator/syncer/syncer.go

```diff
@@ -0,0 +1,336 @@
+	}
+
+	klog.Info("Cluster resource importer started")
+
+	return nil
+}
+
+// createResourceOnDestinationCluster creates the resource on the destination cluster.
+func (s *Service) createResourceOnDestinationCluster(
+	ctx context.Context,
+	resource *unstructured.Unstructured,
+) error {
+	// Extract the GroupVersionResource from the Unstructured object
+	gvk := resource.GroupVersionKind()
+	gvr, err := s.findGVRForGVK(gvk)
+	if err != nil {
+		return err
+	}
+
+	// Namespaces resources should be created within the namespace defined in the Unstructured object
+	namespace := resource.GetNamespace()
+
+	// Run the validating function for the resource.
+	if validatingFn, ok := s.validatingFunctions[gvr]; ok {
+		if ok, err := validatingFn(ctx, resource, s.clients, Add); !ok || err != nil {
+			return err
+		}
+	}
+
+	// When creating a resource on the destination cluster, we must remove the metadata such as UID and Generation.
+	// It's done for all resources.
+	resource = removeMetadata(resource)
+
+	// Run the mutating function for the resource.
+	if mutatingFn, ok := s.mutatingFunctions[gvr]; ok {
+		resource, err = mutatingFn(ctx, resource, s.clients, Add)
+		if err != nil {
+			return xerrors.Errorf("failed to mutate resource: %v", err)
+		}
+	}
+
+	// Create the resource on the destination cluster using the dynamic client
+	_, err = s.clients.destDynamicClient.Resource(gvr).Namespace(namespace).Create(
+		ctx,
+		resource,
+		metav1.CreateOptions{},
+	)
+	if err != nil {
+		return xerrors.Errorf("failed to create resource: %v", err)
+	}
+
+	return nil
+}
+
+func (s *Service) updateResourceOnDestinationCluster(
+	ctx context.Context,
+	resource *unstructured.Unstructured,
+) error {
+	// Extract the GroupVersionResource from the Unstructured object.
+	gvk := resource.GroupVersionKind()
+	gvr, err := s.findGVRForGVK(gvk)
+	if err != nil {
+		return err
+	}
+
+	// Namespaces resources should be created within the namespace defined in the Unstructured object.
+	namespace := resource.GetNamespace()
+
+	// Run the validating function for the resource.
+	if validatingFn, ok := s.validatingFunctions[gvr]; ok {
+		if ok, err := validatingFn(ctx, resource, s.clients, Update); !ok || err != nil {
+			return err
+		}
+	}
+
+	// Run the mutating function for the resource.
+	if mutatingFn, ok := s.mutatingFunctions[gvr]; ok {
+		resource, err = mutatingFn(ctx, resource, s.clients, Update)
+		if err != nil {
+			return xerrors.Errorf("failed to mutate resource: %v", err)
+		}
+	}
+
+	// Create the resource on the destination cluster using the dynamic client
+	_, err = s.clients.destDynamicClient.Resource(gvr).Namespace(namespace).Update(
+		ctx,
+		resource,
+		metav1.UpdateOptions{},
+	)
+	if err != nil {
+		return xerrors.Errorf("failed to create resource: %v", err)
+	}
+
+	return nil
+}
+
+// removeMetadata removes the metadata from the resource.
+func removeMetadata(resource *unstructured.Unstructured) *unstructured.Unstructured {
```

Comment:
The method name `removeMetadata` might be misleading, as readers could interpret it as removing all metadata of the passed resource. Perhaps consider a diffirent name?
```suggestion
func removeUnnecessaryMetadata(resource *unstructured.Unstructured) *unstructured.Unstructured {
```